File: src/Winium.Elements.Desktop/Extensions/ByExtensions.cs
namespace Winium.Elements.Desktop.Extensions
{
    #region using

    using System.Linq;
    using System.Text.RegularExpressions;

    using OpenQA.Selenium;

    #endregion

    public static class ByExtensions
    {
        #region Constants

        private const string DescriptionRegexp = @"\.(.*)\: (.*)";

        #endregion

        #region Public Methods and Operators

        public static object GetStrategy(this By by)
        {
            var match = Regex.Match(by.ToString(), DescriptionRegexp);
            return Regex.Replace(match.Groups[1].Value, "([A-Z])", " $1").Split('[').First().Trim().ToLower();
        }

        public static object GetValue(this By by)
        {
            var match = Regex.Match(by.ToString(), DescriptionRegexp);
            return match.Groups[2].Value;
        }

        #endregion
    }
}

File: java/src/main/java/winium/elements/desktop/extensions/ByExtensions.java
package winium.elements.desktop.extensions;

import org.openqa.selenium.By;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class ByExtensions {
    private ByExtensions() { }

    private static Pattern descriptionRegexp = Pattern.compile("\\.(.*): (.*)");

    public static Object getStrategy(By by) {
        Matcher m = descriptionRegexp.matcher(by.toString());
        if (!m.find())
            return null;
        return m.group(1).replaceAll("([A-Z])", " $1").split("\\[")[0].trim().toLowerCase();
    }

    public static Object getValue(By by) {
        Matcher m = descriptionRegexp.matcher(by.toString());
        if (!m.find())
            return null;
        return m.group(2);
    }
}
